src/components/__tests__/App.test.js

```diff
@@ -0,0 +1,45 @@
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { BrowserRouter } from 'react-router-dom';
+import store from '../redux/configureStore';
+import App from '../../App';
+
+const MockApp = () => (
+  <Provider store={store}>
+    <BrowserRouter>
+      <App />
+    </BrowserRouter>
+  </Provider>
+);
+
+describe('Check if App Component renders correctly', () => {
+  test('renders heading', () => {
+    render(<MockApp />);
+    const headingElement = screen.getByRole('heading', {
+      name: /who's that pokemon\?/i,
+    });
+    expect(headingElement).toBeInTheDocument();
+  });
+
+  test('renders textbox', () => {
+    render(<MockApp />);
+    const textBoxElement = screen.getByRole('textbox');
+    expect(textBoxElement).toBeInTheDocument();
+  });
+
```

Comment:
- [ ] Great job working on the test but I think the tests are failed here after running `npm run test` all the tests didn't passed so kindly check the errors and try to solve them to manage test your api and your components properly ✨🚀🔥🔥🔥